projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -43,8 +43,45 @@
 				return;
 			}
 
+			// Adjust the canvas size.
+			canvas.width = window.innerWidth;
+			canvas.height = window.innerHeight;
+
+			var imageRatio = imgEl.naturalHeight / imgEl.naturalWidth;
+			var winRatio = window.innerHeight / window.innerWidth;
+
+			// Calculate adjusted image dimensions to cover the canvas.
+			var imageHeight = imgEl.naturalHeight;
+			var imageWidth = imgEl.naturalWidth;
+			if ( imageRatio > winRatio ) {
+				imageHeight = imgEl.naturalWidth * winRatio;
+			} else {
+				imageWidth = imgEl.naturalHeight / winRatio;
+			}
+
+			// Apply image blur.
+			if ( context.filter !== undefined ) {
+				context.filter = 'blur(20px) ';
+			} else {
+				// Class used to apply CSS blur on Safari, where it is not supported by the canvas.
+				// Canvas blur is preferred for performance reasons on Chrome.
+				document
+					.querySelector( '.jp-carousel' )
+					.classList.add( 'jp-carousel-canvas-blur-unsupported' );
+			}
+
 			context.filter = 'blur(20px) ';
```

Comment:
This is just a personal observation: sometimes the background image distracts me from the subject image. 

I guess it depends on the contrast and other image variables.


I was fooling around with `context.globalAlpha = 0.5;` to see if I could mitigate this even further. Maybe worth a look if we go ahead.

Reply:
I agree with this, particularly in viewports/image sizes where there's a lot of visible background space. I did try tweaking the blur values as well, but it's a fine line; if the blur is too strong you rapidly approach just having a solid color background, but with a lot more work on the backend 😆

I did not think of using `context.globalAlpha`, that looks like a great option if we go ahead with this!